src/state/opamSwitchState.mli

```diff
@@ -253,8 +253,10 @@ val unavailable_reason:
   'a switch_state -> ?default:string -> name * OpamFormula.version_formula ->
   string
 
-(** Returns whether or not the package can be upgraded to a version tagged with avoid-version *)
-val can_upgrade_to_avoid_version : OpamPackage.Name.t -> 'a switch_state -> bool
+(** Returns [true] when the package has the [avoid-version] flag and there isn't
+    already a version with that flag installed (which disables the
+    constraint) *)
+val avoid_version : 'a switch_state -> package -> bool
```

Comment:
Since this PR also changes the code for the handling of the `avoid-version` flag, i just noticed there wasn’t any tests for this flag. Could you add some if you have some time?

Reply:
I moved some code around and replaced a Set.fold+exists by a filter for efficiency, but that should be functionnally equivalent; indeed there is no fundamental reason for this to be in this PR, I just stumbled upon it while I was writing the PR.

You're right about tests — though I'd prefer not to stall this and add them in another PR.